autotest/utilities/test_gdalalg_raster_select.py

```diff
@@ -0,0 +1,96 @@
+#!/usr/bin/env pytest
+# -*- coding: utf-8 -*-
+###############################################################################
+# Project:  GDAL/OGR Test Suite
+# Purpose:  'gdal raster select' testing
+# Author:   Even Rouault <even dot rouault @ spatialys.com>
+#
+###############################################################################
+# Copyright (c) 2025, Even Rouault <even dot rouault at spatialys.com>
+#
+# SPDX-License-Identifier: MIT
+###############################################################################
+
+import pytest
+
+from osgeo import gdal
+
+
+def get_select_alg():
+    reg = gdal.GetGlobalAlgorithmRegistry()
+    raster = reg.InstantiateAlg("raster")
+    return raster.InstantiateSubAlgorithm("select")
+
+
+def test_gdalalg_raster_select(tmp_vsimem):
+
+    out_filename = str(tmp_vsimem / "out.tif")
+
+    alg = get_select_alg()
+    assert alg.ParseRunAndFinalize(
+        [
+            "--band=3,2,1,mask",
+            "../gcore/data/rgbsmall.tif",
+            out_filename,
+        ],
+    )
+
+    with gdal.OpenEx(out_filename) as ds:
+        assert ds.RasterCount == 4
+        assert [ds.GetRasterBand(i + 1).Checksum() for i in range(4)] == [
+            21349,
+            21053,
+            21212,
+            30658,
+        ]
+
+
+def test_gdalalg_raster_select_mask(tmp_vsimem):
+
+    out_filename = str(tmp_vsimem / "out.tif")
+
+    alg = get_select_alg()
+    assert alg.ParseRunAndFinalize(
+        [
+            "--band=1",
+            "--mask=1",
+            "../gcore/data/byte.tif",
+            out_filename,
+        ],
+    )
+
+    with gdal.OpenEx(out_filename) as ds:
+        assert ds.RasterCount == 1
+        assert ds.GetRasterBand(1).Checksum() == 4672
+        assert ds.GetRasterBand(1).GetMaskBand().Checksum() == 4873
```

Comment:
From the doc I would have expected that band 1 gets used as the mask band, so the checksums would be the same. It looks like Band 1 is used to identify valid pixels, which are then used to generate a mask band with 255 for valid pixels? If this is the case it might be good to test with a mask band that does not result in all pixels being valid.

Reply:
Test modified to check that we have both 0 and 255 values